master/internal/api/logs.go

```diff
@@ -130,6 +125,10 @@ func (l *LogStoreProcessor) Receive(ctx *actor.Context) error {
 			}
 			actors.NotifyAfter(ctx, 500*time.Millisecond, tick{})
 		default:
+			if l.ctx.Err() != nil {
```

Comment:
Saw a situation occasionally where the ctx was checked so far from the send where we sent on a dead gRPC channel, so I moved it up, to where it matters.

Reply:
This `ctx` from the gRPC request. If the stream has closed, `ctx.Err()` is non nil and we stop. We were previously checking it before we fetched the logs. Where it was we were really likely to check, fetch logs and send on a dead channel then check at the right time (since most of the time is spent in fetching, a change usually happens while we're doing that). Though now that I've typed it out, I realize where it is we _always_ do the fetch, even if we're canceled, and then just throw it away instead of erroring, which is wasteful. So I'm going to put it both places, one to avoid wasting work, another to avoid bloating the logs with "failed to send on yada yada.."